packages/byods/src/Logger/index.ts

```diff
@@ -0,0 +1,208 @@
+/* eslint-disable valid-jsdoc */
+import {BYODS_PACKAGE_NAME} from '../constants';
+import ExtendedError from '../Errors/catalog/ExtendedError';
+import {LOGGING_LEVEL, LogContext, LOGGER, LOG_PREFIX} from './types';
+
+/*
+ * These are the order of log levels :-
+ *  error - 1
+ *  warn -  2
+ *  log -   3
+ *  info -  4
+ *  trace - 5
+ *
+ *  Where log level n denotes that level 1 -> level n will be logged.
+ */
+
+let currentLogLevel = LOGGING_LEVEL.error;
+
+/**
+ * A wrapper around console which prints to stderr or stdout
+ * based on the level defined.
+ *
+ * @param message - Log Message to print.
+ * @param level -  Log level.
+ */
+const writeToConsole = (message: string, level: LOGGER) => {
```

Comment:
**NOTE**, I feel this should ideally have been a proxy class like setup... but since we are short on time decided to retain this logic. It works well enough.